File: ServiceCore/Translators/StatusTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class StatusTranslator
    {
        public static ErrorCode TranslateEntityToBO(ProjectStatus _entity, ProjectStatusBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.StatusId;
            bo.Name = _entity.StatusName;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(ProjectStatus _entity, ProjectStatusBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.StatusName = bo.Name;
            return ErrorCode.Success;
        }
    }
}

File: Service/Translators/StatusTranslator.vb
Imports BO
Imports DAL

Public Class StatusTranslator
    Public Shared Function TranslateEntityToBO(_entity As ProjectStatus, bo As ProjectStatusBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.StatusID
        bo.Name = _entity.StatusName
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As ProjectStatus, bo As ProjectStatusBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.StatusName = bo.Name
        Return ErrorCode.Success
    End Function
End Class
